crates/proof-of-sql/src/sql/proof_exprs/and_expr.rs

```diff
@@ -43,26 +43,27 @@ impl ProofExpr for AndExpr {
     #[tracing::instrument(name = "AndExpr::result_evaluate", level = "debug", skip_all)]
     fn result_evaluate<'a, S: Scalar>(
         &self,
-        table_length: usize,
         alloc: &'a Bump,
-        accessor: &'a dyn DataAccessor<S>,
+        table: &Table<'a, S>,
     ) -> Column<'a, S> {
-        let lhs_column: Column<'a, S> = self.lhs.result_evaluate(table_length, alloc, accessor);
-        let rhs_column: Column<'a, S> = self.rhs.result_evaluate(table_length, alloc, accessor);
+        let lhs_column: Column<'a, S> = self.lhs.result_evaluate(alloc, table);
+        let rhs_column: Column<'a, S> = self.rhs.result_evaluate(alloc, table);
         let lhs = lhs_column.as_boolean().expect("lhs is not boolean");
         let rhs = rhs_column.as_boolean().expect("rhs is not boolean");
-        Column::Boolean(alloc.alloc_slice_fill_with(table_length, |i| lhs[i] && rhs[i]))
+        Column::Boolean(
+            alloc.alloc_slice_fill_with(table.num_rows().unwrap_or(0), |i| lhs[i] && rhs[i]),
```

Comment:
I'm not convinced that `0` is the correct default.
If `table` has no columns, it feels like `lhs_column` and `rhs_column` will have to be literals.
Eventually, we should get a proper solution with `ColumnarValue`, but for now, I think it makes more sense to pull the length from the `lhs_column` and `rhs_column` (and assert that the lengths equal).

This is also what is done in `prover_evaluate`.